azure/scope/strategies/machinepool_deployments/machinepool_deployment_strategy.go

```diff
@@ -233,11 +234,16 @@ func getFailedMachines(machinesByProviderID map[string]infrav1exp.AzureMachinePo
 	return machines
 }
 
+// getDeletingMachines is responsible for identifying machines whose VMs are in an active state of deletion
+// but whose corresponding AzureMachinePoolMachine resource has not yet been marked for deletion.
 func getDeletingMachines(machinesByProviderID map[string]infrav1exp.AzureMachinePoolMachine) []infrav1exp.AzureMachinePoolMachine {
 	var machines []infrav1exp.AzureMachinePoolMachine
 	for _, v := range machinesByProviderID {
-		// ready status, with provisioning state Succeeded, and not marked for delete
-		if v.Status.ProvisioningState != nil && *v.Status.ProvisioningState == infrav1.Deleting {
+		if v.Status.ProvisioningState != nil &&
+			// provisioning state is Deleting
+			*v.Status.ProvisioningState == infrav1.Deleting &&
+			// If we've already marked the DeletionTimestamp then we have no work to do
+			v.DeletionTimestamp.IsZero() {
```

Comment:
@devigned so I've combined the if statements into one here, but you seemed to suggest that this would be improved if the "Azure VM is actively being deleted" + "capz has already marked this ampm for deletion" are OR'd together (instead of AND'ed). Can you clarify that subtlety, if you think we should make it OR?